addons/portal/views/portal_templates.xml

```diff
@@ -21,6 +21,7 @@
                 <t t-call-assets="web.assets_frontend" t-css="false"/>
             </t>
             <t t-set="head" t-value="head_website + (head or '')"/>
+            <t t-set="wrapwrap_classes" t-value="'o_portal_bg_dark'"/>
```

Comment:
I think the only way to do this properly is that way:
- Remove the `wrapwrap_classes` system
- Add a conditional class on the #wrapwrap element: `#{is_portal ? 'o_portal' : ''}`
- Mark all portal pages as portal ones by adding `<t t-set="is_portal" t-value="True"/>` after each `<t t-call="portal.frontend_layout"/>`

Doing this will also solve the potential JS problem you told me.